presto-spi/src/main/java/com/facebook/presto/spi/function/PropagateSourceStats.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.function;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public enum PropagateSourceStats
+{
+    MAX,
+    SUM,
+    SOURCE_STATS,
+    ROW_COUNT,
+    ROW_COUNT_TIMES_INV_NULL_FRACTION, // row_count * (1 - null_fraction)
+    TYPE_WIDTH_VARCHAR,
+    MAX_TYPE_WIDTH_VARCHAR,
+    UNKNOWN;
+
+    /*
+     * Stats are simple when their value is calculated by operating on stats from source stats of the argument which
+     * is annotated.
+     */
+    public static final Set<PropagateSourceStats> SIMPLE_STATS =
+            new HashSet<>(Arrays.asList(UNKNOWN, ROW_COUNT, ROW_COUNT_TIMES_INV_NULL_FRACTION, TYPE_WIDTH_VARCHAR, SOURCE_STATS));
```

Comment:
For this one, we would have to add guava dependency to presto-spi, I was not too sure the kind of implication that would have. Historically, many java package/projects have shaded guava dependency because the library they depended on has a version mismatch.
@ZacBlanco once hesitated to do it. Do you think we can do it?

Reply:
I agree we should avoid adding guava into the SPI. However, I think this should be wrapped with a `Collections.unmodifiableMap` at least to prevent the map from changing

Reply:
Didn't realize this was a SPI class, makes sense